.contrib/Harvesters/WoWTools Harvester/Harvester.py

```diff
@@ -22,193 +24,170 @@
-                nums[i], nums[i + 1] = nums[i + 1], nums[i]
-                swapped = True
-
-# This function takes the input(Latest Build ex. "10.0.0.44500") and add it to all the BuildList files at the end.
-def add_latest_build(build):
-    data_folder = "C:\\Program Files (x86)\\World of Warcraft\\_retail_\\Interface\\AddOns\\AllTheThings\\.contrib\\Harvesters\\WoWToolsHarvester\\BuildLists\\"
-    for thing in thing_list:
-        path = data_folder + "BuildList "+ thing + ".txt"
-        with open(path, "a") as build_list:
-            build_list.write(build+"\n")
-
-# This function takes the inputs(thing from thing_list, build ex. "10.0.0.44500"). Creates a csv file and from that csv file only choose specific columns of information(usually useful IDs) and writes them in a new file.
-def get_tools_IDs(thing, build):
-    if thing == "Achievements":
-        thing_url = "achievement"
-    elif thing == "Factions":
-        thing_url = "faction"
-    elif thing == "Flight Paths":
-        thing_url = "taxinodes"
-    elif thing == "Followers":
-        thing_url = "garrfollower"
-    elif thing == "Illusions":
-        thing_url = "transmogillusion"
-    elif thing == "Mounts":
-        thing_url = "mount"
-    elif thing == "Pets":
-        thing_url = "battlepetspecies"
-    elif thing == "Quests":
-        thing_url = "questv2"
-    #elif thing == "Recipes":
-        #thing_url = "skilllineability"
-    elif thing == "Titles":
-        thing_url = "chartitles"
-    elif thing == "Toys":
-        thing_url = "toy"
-    elif thing == "Transmog":
-        thing_url = "itemmodifiedappearance"
-    url = "https://wow.tools/dbc/api/export/?name={thing_url}&build={build}"
+
+class Things(Enum):
+    Achievements = auto()
+    Factions = auto()
+    FlightPaths = auto()
+    Followers = auto()
+    Illusions = auto()
+    Mounts = auto()
+    Pets = auto()
+    Quests = auto()
+    Recipes = auto()
+    Titles = auto()
+    Toys = auto()
+    Transmog = auto()
+
+
+def add_latest_build(build: str) -> None:
+    """Append the latest build to all the BuildList files."""
+    for thing in Things:
+        with open(Path("BuildLists", f"BuildList{thing.name}.txt"), "a") as build_list:
+            build_list.write(build + "\n")
+
+
+def get_thing_ids(thing: Things, build: str) -> list[str]:
+    """Get the IDs of a thing from a build."""
+    thing2table = {
+        Things.Achievements: "achievement",
+        Things.Factions: "faction",
+        Things.FlightPaths: "taxinodes",
+        Things.Followers: "garrfollower",
+        Things.Illusions: "transmogillusion",
+        Things.Mounts: "mount",
+        Things.Pets: "battlepetspecies",
+        Things.Quests: "questv2",
+        Things.Recipes: "skilllineability",
+        Things.Titles: "chartitles",
+        Things.Toys: "toy",
+        Things.Transmog: "itemmodifiedappearance",
+    }
+    url = f"https://wow.tools/dbc/api/export/?name={thing2table[thing]}&build={build}"
     headers = {
```

Comment:
Do we need headers?

Reply:
No, not that I recall

Reply:
Yeah I forgot that we wanted items also right?